### PR TITLE
SB-16725 - ItemSet API Spec

### PR DESCRIPTION
**Pull request by maheshkumargangula**
@Basreena @NupurSancheti - This pull-request is for release-2.7.0

### CODE REVIEW

**Comment by keshavprasadms** [comment]
Hello @maheshkumargangula,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://550.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://550.qa.docs.sunbird.org/2.6.0/apis/content// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br />  <br/> **Modified sections are -** <br />

**Comment by keshavprasadms** [comment]
Hello @maheshkumargangula,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://550.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://550.qa.docs.sunbird.org/2.6.0/apis/content// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br />  <br/> **Modified sections are -** <br />

**Comment by keshavprasadms** [comment]
Hello @maheshkumargangula,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://550.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://550.qa.docs.sunbird.org/2.6.0/apis/content// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br />  <br/> **Modified sections are -** <br />

File: apis/itemsetapi/itemsetapi.yaml [hunk 1]
```diff
@@ -0,0 +1,840 @@
...
+          
+           The body is the representation of the resource object to create an item set (an array of parameters that describes the item set)
+          schema:
+            $ref: '#/definitions/CreateItemSetRequest'
+        - name: Content-Type
+          in: header
+          required: true
+          type: string
+          description:  >-
+            The ItemSet Type entity is the media type of the resource.Possible media types can be:-
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: should this be Content type, not ItemSet Type, as that is what is mentioned against the name parameter?

File: apis/itemsetapi/itemsetapi.yaml [hunk 2]
```diff
@@ -0,0 +1,840 @@
...
+          required: true
+          type: string
+          description:  >-
+            The ItemSet Type entity is the media type of the resource.Possible media types can be:- 
+              - application/json
+        - name: x-channel-id
+          in: header
+          required: true
+          type: string
+          description: 'Represents a physical location (e.g: state) uniquely to which the user belongs.'
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: do you mean the tenant ID? Why does it need to be physical location?

File: apis/itemsetapi/itemsetapi.yaml [hunk 3]
```diff
@@ -0,0 +1,840 @@
...
+          
+            The body is the representation of the resource object for updating the item set.
+          schema:
+            $ref: '#/definitions/UpdateItemSetRequest'
+        - name: Content-Type
+          in: header
+          required: true
+          type: string
+          description: >-
+            The ItemSet Type entity is the media type of the resource.Possible media types can be:-
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: should this be Content type, not ItemSet Type, as that is what is mentioned against the name parameter?

File: apis/itemsetapi/itemsetapi.yaml [hunk 4]
```diff
@@ -0,0 +1,840 @@
...
+        - name: X-Authenticated-Userid
+          in: header
+          required: true
+          type: string
+          description: 'Represents the User ID authorized to execute the API.'
+        - name: X-Channel-id
+          in: header
+          required: true
+          type: string
+          description: 'Represents a physical location (e.g: state) uniquely to which the user belongs.'
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: do you mean the tenant ID? Why does it need to be a physical location? pls check this for all X-Channel-id parameters in the API

File: apis/itemsetapi/itemsetapi.yaml [hunk 5]
```diff
@@ -0,0 +1,840 @@
...
+        - name: Body
+          in: body
+          required: true
+          description: >-
+           The body parameter refers to the format of the request, the request body (payload) takes the following media types to make the payload:'
+            
+            - JSON 
+            - XML data
+          
+            The body is the representation of the resource object for deleting item set
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: does retiring the ItemSet = deleting the ItemSet? this name for this endpoint is retire , not delete. Hence please make it consistent.

File: apis/itemsetapi/itemsetapi.yaml [hunk 6]
```diff
@@ -0,0 +1,840 @@
...
+          description: 'Represents the User ID registered to execute the API.'
+        - name: Authorization
+          in: header
+          required: true
+          type: string
+          description: 'To make use of the API, you require authorization. Raise a request to the administrator for the use of the API. You will receive the authorization key. Specify the key received, here.'
+      responses:
+        '200':
+          description: >-
+            SUCCESS. The **Delete ItemSet** operation was successful!
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: The name of the operation ** Delete ItemSet** is not consistent with the name of the API end point. It should be consistent.

File: apis/itemsetapi/itemsetapi.yaml [hunk 7]
```diff
@@ -0,0 +1,840 @@
...
+          schema:
+            $ref: '#/definitions/ItemSetCreateResponse'
+        '400':
+          description: >-
+            BAD REQUEST. The **Delete ItemSet** operation failed. You may have missed input for a mandatory parameter.
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: The name of the operation ** Delete ItemSet** is not consistent with the name of the API end point. It should be consistent.

File: apis/itemsetapi/itemsetapi.yaml [hunk 8]
```diff
@@ -0,0 +1,840 @@
...
+          schema:
+            $ref: '#/definitions/ItemSetCreateErrorResponse'
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: Please check the name of the schema ref.

File: apis/itemsetapi/itemsetapi.yaml [hunk 9]
```diff
@@ -0,0 +1,840 @@
...
+        '500':
+          description: >-
+            'INTERNAL SERVER ERROR. We track these errors automatically and try to set it right at the earliest. Try refreshing the page. If the problem persists contact us at info@sunbird.org.'
+          schema:
+            $ref: '#/definitions/ItemSetCreateErrorResponse'
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: Please check the name of the schema ref.

File: apis/itemsetapi/itemsetapi.yaml [hunk 10]
```diff
@@ -0,0 +1,840 @@
...
+          in: header
+          required: true
+          type: string
+          description: 'To make use of the API, you require authorization. Raise a request to the administrator for the use of the API. You will receive the authorization key. Specify the key received, here.'
+      responses:
+        '200':
+          description: >-
+            SUCCESS. The **Delete ItemSet** operation was successful!
+          schema:
+            $ref: '#/definitions/ItemSetCreateResponse'
```
Comment (Sunitashelar) [inline apis/itemsetapi/itemsetapi.yaml]: Please check the name of the schema ref.

**Comment by Sunitashelar** [comment]
@keshavprasadms please address the review comments for ItemSet API

**Comment by keshavprasadms** [comment]
Hello @maheshkumargangula,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://550.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://550.qa.docs.sunbird.org/2.6.0/apis/content// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br />  <br/> **Modified sections are -** <br />

**Comment by keshavprasadms** [comment]
Hello @maheshkumargangula,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://550.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://550.qa.docs.sunbird.org/2.6.0/apis/content// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br /> - http://550.qa.docs.sunbird.org/2.6.0/apis/itemsetapi// <br />  <br/> **Modified sections are -** <br />

**Pull request merged**